### ISSUE TITLE
Windows display blank on reboot after Windows 11 ARM preview/24H2 update 

### ISSUE THREAD

**Issue by chadlwilson** [utmapp/UTM#6332]
**Describe the issue**  

After an update to Windows 11 ARM Insider Preview `26200.5001` (similar for those updating to official release `24H2`) when the VM rebooted the screen was black/blank. The VM seemed to be doing stuff underneath.

I could **workaround/fix this** by
- Changing the display device from `virtio-ramfb-gl` (default, I never have changed this) to `virtio-gpu-gl-pci`
- Restarted VM. Display now works.
- Reinstalling UTM/Spice guest tools (this is normal to need to do after an upgrade due to the way insider previews work)
- Changing back to `virtio-ramfb-gl`. Display still works.

I'm not sure if this is a change on UTM (from `4.5.x`) or Windows 11 ARM Insider preview nor what the differences are between the display drivers, but recording it here in case others have the same issue and need a workaround.

Feel free to close this if it's expected.

![image](https://github.com/utmapp/UTM/assets/29788154/8d94d323-a6b8-4340-b2cd-c6997dd36b35)
![image](https://github.com/utmapp/UTM/assets/29788154/9216fc90-1876-4a8a-9496-d12d8c5ab0d4)

**Configuration**  
* UTM Version: `4.5.2`
* macOS Version: Sonoma `14.4.1`
* Mac Chip (Intel, M1, ...): `M1`

**Debug log**  
I don't think I can recreate this now - perhaps at the next Insider Preview update? :-(

**Upload VM**  
[config.plist.zip](https://github.com/utmapp/UTM/files/15230946/config.plist.zip)

**Comment by chadlwilson** [comment]
For what it's worth, I didn't have the same problem after updating to `26212.5000` so not sure if this was something to do with that release; or the QEMU/UTM update.

**Comment by chadlwilson** [comment]
I did another Insider Preview upgrade and did not have an issue with the display. Will close this for now as perhaps it was something quirky specific to me setup, and I am using Insider Preview builds so it's possible this is specific to that, rather than a UTM issue.

Can re-open if others have similar issues.

**Comment by Kaiojag** [comment]
Only now I have the same problema as your. Even changing the display option I still have a black screen. I also do not know how  to reinstall UTM/Spice guest tools how can I do it? I am sorry for reopening the thread.

**Comment by chadlwilson** [comment]
Well you can't reinstall the spice tools until you can actually see the screen and interact with the guest, so need to find a way to fix that first.

There could probably be many reasons the display isn't working, this ticket was specifically about it happening as an unexpected side effect of a Window update/upgrade (which hasn't happened to me before or since).

**Comment by halter73** [comment]
I noticed the same issue shortly after updating to [Windows 11 Insider Preview Build 26120.1252](https://blogs.windows.com/windows-insider/2024/07/15/announcing-windows-11-insider-preview-build-26120-1252-dev-channel/). It might have happened earlier without me noticing since I typically RDP in, but I appreciate the tip that you need to reinstall UTM/Spice guest tools to fix this issue. That worked for me. It's good to know if I lose RDP for some reason, I can still use the UTM display.

It might be worth reopening this issue for visibility especially if changing the display option doesn't work around the issue. I found https://www.reddit.com/r/mac/comments/1dk7z45/windows_11_vm_on_utm_black_screen_after_software/ before finding this which suggests that Windows updates are causing the UTM display not to function and just show a black screen. I'm glad I was able to fix the issue without reverting the update.

**Comment by chadlwilson** [comment]
Interesting - so it isn't/wasn't just me...

Since I was on the canary channel, if it's something from a windows update it does stand to reason that I'd be hit first.

I haven't done any updates for a while as I'm trying to get off the canary channel at some point without a complete reinstall, so had switched to Dev and hoping for the build #s to catch up for a small window some time this year 😅

**Comment by RealDyllon** [comment]
I'm experiencing this issue as well. I'm able to see the login screen, but upon logging in the screen turns black.

**Comment by corwin-of-amber** [comment]
@chadlwilson's symptom occurred for me as well after updating to 240927. Switching to `virtio-gpu-gl-pci` and reinstalling guest tools worked as prescribed.

BTW any reason not to leave the display setting as `virtio-gpu-gl-pci`? In case future updates will reintroduce this problem.

**Comment by msart-it-vision** [comment]
Having Updated my UTM VM Windows 11 from 22H2 with the 24H2 from ISO and have the BlackScreen also.

Above mentioned Method did not work but the Display change to Option virtio-gpu-gl-pci works.
Only downside it is only 1024x786 and no Screen Sizing or Resolution change possible.

I use RDP and lucky me had it also configured before the Update.

No virtio-ramfb-gl anymore i had used from the beginning in mid 2022.

MBA M2 MacOS 15 - UTM 4.5.4

**Comment by chadlwilson** [comment]
I suspect the particular update is causing an issue regardless of the display device - since it's possible to change it temporarily and then change it back to the original driver without ongoing issue.

**Comment by msart-it-vision** [comment]
Have tried the above method  changing device several times but as i stated that did not work.
And yes, i have always reinstalled the Spice Tools.
The GraficCard shows up in the devicemanager but on the Mac Side it only flashes shortly when entering Remote Desktop.

Windows 11 Pro 24H2 26100.1882
[KB5043178] MBA M2 MacOS 15 - UTM 4.5.4

**Comment by darianss** [comment]
Hi all, this didn't really work for me. Switching the display device to ```virtio-gpu-gl-pci``` still leads to a black screen, after the initial Windows loading screen showing up fine. Not sure what else to try.

**Comment by osy** [comment]
Duplicate of #6837

**Comment by osy** [comment]
There is definitely something that broke in Windows. Another issue is that when you try to install Windows 24H2 ARM64, it will load the installer and then immediately show "Guest has not initialized the display (yet)" when the graphics drivers are loaded. This can be worked around by un-mounting the guest tools ISO and manually installing it afterwards.

**Comment by denischernenko** [comment]
I experienced "Display Output Not Active" error too. Could not fix it, but found temporary solution: 

1. Skip installing drivers from \Drivers\viogpudo and change — this is how I can still operate the VM. I just made a little batch script to bypass it.

2. Emulated Display Card in VM Hardware settings -> virtio-ramfb-gl (GPU Not Supported).

Though, I still see 'Display Output Not Active' for a second or two while booting/restarting VM, but now it does not obstruct my work anymore.

**Comment by saidatta** [comment]
ditto. I gave up on emulating win 11. will wait until a proper fix.

**Comment by skapsarali** [comment]
getting same error "Display Output Not Active"

**Comment by creechy** [comment]
Just curious if there are any thoughts on a path for this? Could there eventually be a solution without workarounds?

**Comment by timothynoel** [comment]
FWIW, the latest 4.63 release addressed this for me. In 4.62 I was getting blank screens on my windows (23H2) and Debian Bookworm VMs, no matter the display device used. Now everything is working. I see the display not active message very briefly in Debian but everything works fine afterward

**Comment by osy** [comment]
There's two different issues here. #6861 was addressed by 4.6.3 and affects every QEMU VM as soon as you start it. This one only affects Windows 11 24H2 (including after updating to it) and is caused by a Windows driver issue.

**Comment by timothynoel** [comment]
Thanks for the correction, I missed that other issue, and also, thanks for the fix on that one.

**Comment by creechy** [comment]
> Just curious if there are any thoughts on a path for this? Could there eventually be a solution without workarounds?

FWIW - I still had to do the workaround - switch video card - reinstall guest tools - switch video back. 
UTM 6.4.3, M3 Max, Sequoia, Windows 11 ARM (guest)
Keeping a backup of the VM for a while :-)

**Comment by msart-it-vision** [comment]
Well...

i had the Problem related to 24H2 and after booted from UTM 463 with no Display out and connecting via Remote Desktop on my MBA M2 i activated/downloadet the spice tools iso via the GUI Frame of UTM and installed the Driver in Win11 Pro Arm 24H2 again.

Then the Remote Desktop was Black Screen and the GUI of UTM Sand colored.
I had to Hard Reset but jep, after Booting i had the GUI Display out again.

So even if @timothynoel mixed it up, he goz me to try 463. 
My Win 11 Pro for ARM 24H2 Grafic Out Problem is solved after 3 Month with Remote Desktop only.

**Comment by bdutta** [comment]
Glad to find this workaround as it worked for me and unblocked me. The problem (black screen) cropped up after the guest additions were added, so I got the network, Microsoft Market, and a resizable display (I'm assuming that all of those somehow need guest additions). I stopped the VM, then removed / unmounted the guest-additions iso and then the Win11 installer iso, changed the display driver, and restarted VM, then everything seems to work fine.

**Comment by chadlwilson** [comment]
Yeah, there seem to possibly be a few related issues of blank or non-functional display being referred to here which might be slightly different to when I first opened the ticket where it was a transient problem after a specific upgrade that eventually rolled out more widely in 24H2.

Subsequent to this I have switched to permanently using the (slow, laggy) `virtio-ramfb` display device as I found with the GPU variants that my display would be permanently dead when swapping between an external device and laptop display with a running VM, usually after sleep. It was too annoying so I swapped, as I prefer not to use remote desktop to access my Win VM. 

But that's another example of a separate blank display device problem that might be a bit independent of this one.

**Comment by degen91** [comment]
The display driver included in the newer virtio-win package (0.1.266 vs 0.1.229) is nearly two years newer than the driver included with UTM's version of Guest Tools, and is tagged 100.100.104.26600, perhaps indicating it has compatibility with 24H2. Other drivers which have seen updates are net, balloon, RNG, and serial.

**Comment by osy** [comment]
The same issue exists on the newer drivers as well.

**Comment by aidangoettsch** [comment]
I just did a clean install of Windows 11 24H2 and installed SPICE immediately after. Similar to #6853, while installing SPICE it seems like the new driver is initialized and I get "Display output not active". 

It seems like this bricks the VM after rebooting. I've tried all the virtio devices available, virtio-gpu-device gives "Display output not active" right on boot, virtio-gpu-gl-device just gives a blank window right on boot, the rest of the virtio-gpu options hang on the UTM logo showing "Start boot option". and the virtio-ramfb ones black screen.

Is there a workaround here or do I just need to install an older build and block it from updating?

**Comment by osy** [comment]
Fix will be coming soon. In the meantime, follow the workaround in the first post.

**Comment by aidangoettsch** [comment]
For anyone else with the same issue, it turns out the trick to getting the VM to show video was to remove the ISOs for the installer and SPICE. That resolved the issues with virtio-gpu-gl-pci and allowed me to follow that workaround.

**Comment by KAGayan** [comment]
Thanks @aidangoettsch ejecting the Windows ISO and the Tools fix the issue for me.

**Comment by louis49** [comment]
Same issue on installing Win11_24H2_French_Arm64.iso or 26100.2033.241004-2336.ge_release_svc_refresh_CLIENTCONSUMER_RET_A64FRE_fr-fr.iso with latest version (4.6.4) on macOS 15.3

**Comment by louis49** [comment]
It's not about updating an existing previously working VM but start a fresh new install

**Comment by chadlwilson** [comment]
Yes, I understand that - better to create a new issue if you have a problem that is different than that originally raised, rather than comment on an old, closed issue which relates to something different. Doesn't help to conflate two different, but related issues.

**Comment by sinwoobang** [comment]
I am installing it third times due to the error. By the way, I do not understand it–what is the connection between the ejection of ISOs and the display recovery?
